src/panels/lovelace/hui-root.ts

```diff
@@ -1208,6 +1273,35 @@ class HUIRoot extends LitElement {
     showShortcutsDialog(this);
   }
 
+  private async _applyUndoRedo(item: UndoStackItem) {
+    this._configChangedByUndo = true;
+    try {
+      await this.lovelace!.saveConfig(item.config);
+    } catch (err: any) {
+      showToast(this, {
```

Comment:
We should reset this flag 

```suggestion
    } catch (err: any) {
      this._configChangedByUndo = false;
      showToast(this, {
```